### ISSUE TITLE
Roslyn ignores [NotNullIfNotNull] annotation on extension deconstruction method calls

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#50443]
**Version Used**: 
```
Branch master (11 Jan 2021)
Latest commit 1223006 by Julien Couvreur:
Disallow setting init-only property in nested initializer (#50200)
```

**Steps to Reproduce**:

Compile the following code
```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

class C
{
  void M(C c)
  {
    (string s1, _) = c;
  }
}

static class Ext
{
  public static void Deconstruct(this C? c, [NotNullIfNotNull("c")] out string? s1, out string? s2) => s1 = s2 = "";
}
```

**Expected Behavior**:
No warnings. The out parameter is known to be non-nullable since `c` is not nullable.

**Actual Behavior**:
False `CS8600: Converting null literal or possible null value to non-nullable type.` is reported for `string s1` in `M`

**Comment by Tragetaschen** [comment]
When you call the `Deconstruct` extension method manually, the nullability annotations are recognized correctly.
